File: Ejemplos/ProgramacionOrientadaObjetos/src/main/java/Cirujano.java
public class Cirujano extends Medico implements ICirujano {
    private final String especialidad;

    public Cirujano(String nombre, String numeroColegiado, String especialidad) {
        super(nombre, numeroColegiado); // Debe ser la primera línea
        this.especialidad = especialidad;
    }

    public String getEspecialidad() {
        return especialidad;
    }

    @Override
    public void operar() {
        System.out.println("Cirujano " + getNombre() + " operando");
    }

    @Override
    public void suturar() {
        System.out.println("Cirujano " + getNombre() + " suturando");
    }

    @Override
    public void saludar() {
        System.out.println("Hola, soy el cirujano " + getNombre());
    }
}

File: Ejemplos/Hospital-MVC-Kotlin-Mockk/src/main/kotlin/models/Cirujano.kt
package models

import java.util.*

class Cirujano(
    id: UUID,
    nombre: String,
    val habilidad: String,
) : Medico(id = id, nombre = nombre), ICirujano {
    override fun calculaBonus(): Double {
        return 50.0
    }

    override fun operar() {
        println("Soy el cirujano $nombre y estoy operando")
    }

    override fun toString(): String {
        return "Cirujano(id=$id, nombre='$nombre', habilidad='$habilidad')"
    }

}